src/fmu/dataio/scripts/copy_preprocessed.py

```diff
@@ -0,0 +1,167 @@
+with the contents::
+  WF_COPY_PREPROCESSED_DATAIO <SCRATCH>/<USER>/<CASE_DIR> <CONFIG_PATH> '../../share/preprocessed/'
+
+Add following lines to your ERT config to have the job automatically executed::
+  LOAD_WORKFLOW ../bin/workflows/xhook_copy_preprocessed_data
+  HOOK_WORKFLOW xhook_copy_preprocessed_data PRE_SIMULATION
+"""  # noqa
+
+
+def main() -> None:
+    """Entry point from command line
+
+    When script is called from an ERT workflow, it will be called through the 'run'
+    method on the WfCopyPreprocessedData class. This context is the intended usage.
+    The command line entry point is still included, to clarify the difference and
+    for debugging purposes.
+    """
+    parser = get_parser()
+    commandline_args = parser.parse_args()
+    copy_preprocessed_data_main(commandline_args)
+
+
+class WfCopyPreprocessedData(ErtScript):
+    """A class with a run() function that can be registered as an ERT plugin.
+
+    This is used for the ERT workflow context. It is prefixed 'Wf' to avoid a
+    potential naming collisions in fmu-dataio."""
+
+    def run(self, *args: str) -> None:
+        """Parse arguments and call copy_preprocessed_data_main()"""
+        parser = get_parser()
+        workflow_args = parser.parse_args(args)
+        copy_preprocessed_data_main(workflow_args)
+
+
+def copy_preprocessed_data_main(args: argparse.Namespace) -> None:
+    """Copy the preprocessed data to scratch and upload it to sumo."""
+
+    check_arguments(args)
+    logger.setLevel(args.verbosity)
+
+    searchpath = Path(args.ert_config_path) / args.inpath
+    match_pattern = "[!.]*"  # ignore metafiles (starts with '.')
+    files = [f for f in searchpath.rglob(match_pattern) if f.is_file()]
+    logger.debug("files found %s", files)
+
+    if not files:
+        raise ValueError(f"No files found in {searchpath=}, check spelling.")
+
+    logger.info("Starting to copy preprocessed files to <caseroot>/share/observations/")
+    for filepath in files:
+        ExportPreprocessedData(
+            casepath=args.ert_caseroot,
+            is_observation=True,
+        ).export(filepath)
+        logger.info("Copied preprocessed file %s", filepath)
+
+    logger.debug("copy_preprocessed_data_main.py has finished.")
+
+
+def check_arguments(args: argparse.Namespace) -> None:
+    """Do basic sanity checks of input"""
+    logger.debug("Checking input arguments")
+    logger.debug("Arguments: %s", args)
+
+    if args.global_variables_path:
+        warnings.warn(
+            "The global variables path is no longer needed. Please remove the "
+            "'--global_variables_path' argument and path from the workflow file.",
+            FutureWarning,
+        )
+
+    if not Path(args.ert_caseroot).is_absolute():
+        logger.debug("Argument 'ert_caseroot' was not absolute: %s", args.ert_caseroot)
+        raise ValueError("'ert_caseroot' must be an absolute path")
```

Comment:
`argparse.ArgumentError` is another option here, but I see this is most likely taken from create_case_metadata